Completion/Unix/Command/_stow

```diff
@@ -0,0 +1,89 @@
+#compdef stow chkstow
+
+#
+# A zsh completion script for GNU stow (https://www.gnu.org/software/stow/)
+#
+
+__stow_complete_packages() {
+  local stow_dir=${(Q)1}
+  local -a stow_pkg_list=( $stow_dir/*(-/N:t) )
+
+  if [[ ${#stow_pkg_list} -gt 0 ]]; then
+    _values -C "packages from $stow_dir" ${stow_pkg_list[@]}
+  else
+    _message "no packages found in $stow_dir"
+  fi
+}
+
+case $service in
+  stow)
+    local context state line curcontext="$curcontext" ret=1
+    typeset -A opt_args
+    # Others local variables
+    local stow_dir arguments
+
+    arguments=(
+      '(- *)'{--help,-h}'[show help]'
+      '(- *)'{--version,-V}'[show version number]'
+      '(-d --dir)'{-d+,--dir=}'[set the stow dir (default is current dir)]:stow dir:_files -/'
+      '(-t --target)'{-t+,--target=}'[set the target dir (default is parent of stow dir)]:target dir:_files -/'
+      '*'{-S,--stow}'[stow the package names that follow]:stow package:->stow_package'
+      '*'{-D,--delete}'[unstow the package names that follow]:stow package:->stow_package'
+      '*'{-R,--restow}'[restow (unstow and stow again) the package names that follow]:stow package:->stow_package'
+      '--adopt[adopt already existing plain file]'
+      '--ignore=[ignore files ending with this perl regex]:regexp:'
+      "--defer=[don't stow files beginning with this perl regex]:regexp:"
+      '--override=[force stowing files beginning with this perl regex]:regexp:'
+      '--no-folding[disable any further tree folding or tree refolding]'
+      '(-p --compat)'{-p,--compat}'[use legacy algorithm for unstowing]'
+      '(-n -no --simulate)'{-n,--no,--simulate}'[do not actually make any filesystem changes]'
+      '(-v --verbose)'{-v,--verbose}'[increase or set the level of verbosity (from 0 to 5)]'
```

Comment:
It looks like `--verbose` is supposed to take an optional verbosity level, and it should be possible to use them multiple times so they stack up (e.g., you can do either `-vvv` or `--verbose=3`)

Reply:
Yes, initially I used this (see my first commit 64fcfa8f6) :

    '(-v --verbose)'{-v,--verbose}'[increase verbosity]'
    '(-v --verbose)'--verbose=-'[set verbose level to n]:level:(0 1 2 3 4 5)'`

I don't know if there's a better way to do this

Reply:
If the argument to --verbose is optional, use two colons to indicate so. For example:

    '*-v[increase verbosity]'
    '*--verbose=-[increase verbosity]::level:(0 1 2 3 4 5)'

I'd be inclined to have the same description so the options are grouped together. Things like "to n" seem weird without more context for what "n" is anyway. And if options can be repeated, we want the initial `*` rather than exclusion lists.

Reply:
done in 675da0869